src/main/java/de/tum/cit/aet/artemis/core/repository/cleanup/CustomDataCleanupRepository.java

```diff
@@ -0,0 +1,296 @@
+        entityManager.createQuery("""
+                    DELETE FROM TextBlock tb
+                    WHERE tb.feedback IN (SELECT f FROM Feedback f JOIN f.result WHERE f.result IS NULL)
+                """).executeUpdate();
+        entityManager.createQuery("DELETE FROM Feedback f WHERE f.result IS NULL").executeUpdate();
+        entityManager.createQuery("DELETE FROM StudentScore ps WHERE ps.user IS NULL").executeUpdate();
+        entityManager.createQuery("DELETE FROM TeamScore ps WHERE ps.team IS NULL").executeUpdate();
+        // all long feedback text records that are part of feedback that is part of an orphan result
+        entityManager.createQuery("""
+                    DELETE FROM LongFeedbackText lft
+                    WHERE lft.feedback.id IN (
+                        SELECT f.id
+                        FROM Feedback f
+                        WHERE f.result.participation IS NULL AND f.result.submission IS NULL
+                    )
+                """).executeUpdate();
+        // we cannot delete feedbacks that are part of orphan results because of text block relation
+        entityManager.createQuery("""
+                    DELETE FROM TextBlock tb
+                    WHERE tb.feedback IN (SELECT f FROM Feedback f JOIN f.result r WHERE r.submission IS NULL AND r.participation IS NULL)
+                """).executeUpdate();
+        // finally, we can delete feedbacks that are connected to orphan results
+        entityManager.createQuery("""
+                    DELETE FROM Feedback f
+                    WHERE f.result IN (SELECT r FROM Result r WHERE r.submission IS NULL AND r.participation IS NULL)
+                """).executeUpdate();
+        entityManager.createQuery("""
+                    DELETE FROM Rating rt
+                    WHERE rt.result IN (SELECT r FROM Result r where r.submission IS NULL AND r.participation IS NULL)
+                """).executeUpdate();
+        entityManager.createQuery("DELETE FROM Result r WHERE r.participation IS NULL AND r.submission IS NULL").executeUpdate();
+    }
+
+    @Override
+    public List<Long> getUnnecessaryPlagiarismComparisons(ZonedDateTime deleteFrom, ZonedDateTime deleteTo) {
+        return entityManager.createQuery("""
+                      SELECT pc.id FROM PlagiarismComparison pc
+                      JOIN Submission s1 ON pc.submissionA.submissionId = s1.id
+                      JOIN Submission s2 ON pc.submissionB.submissionId = s2.id
+                      JOIN s1.participation p1
+                      JOIN s2.participation p2
+                      JOIN p1.exercise e1
+                      JOIN p2.exercise e2
+                      JOIN e1.course c1
+                      JOIN e2.course c2
+                      WHERE pc.status = de.tum.cit.aet.artemis.plagiarism.domain.PlagiarismStatus.NONE AND
+                      LEAST(c1.endDate, c2.endDate) < :deleteTo AND
+                      GREATEST(c1.startDate, c2.startDate) > :deleteFrom
```

Comment:
Just to make sure of this is intended?
```suggestion
                      GREATEST(c1.endDate, c2.endDate) < :deleteTo AND
                      LEAST(c1.startDate, c2.startDate) > :deleteFrom
```